src/field/FieldTrackView.hh

```diff
@@ -0,0 +1,104 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file FieldTrackView.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Types.hh"
+#include "base/Macros.hh"
+#include "physics/base/Units.hh"
+#include "field/FieldInterface.hh"
+#include "geometry/GeoTrackView.hh"
+#include "physics/base/ParticleTrackView.hh"
+
+#include <VecGeom/navigation/VNavigator.h>
+#include <VecGeom/navigation/NavigationState.h>
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Read/write view to the field state of a single track.
+ */
+
+class FieldTrackView
+{
+  public:
+    //@{
+    //! Type aliases
+    using NaviState = vecgeom::NavigationState;
+    using Navigator = vecgeom::VNavigator;
+    //@}
+
+  public:
+    // Construct from GeoTrackView and ParticleTrackView
+    inline CELER_FUNCTION FieldTrackView(const GeoTrackView&      gt_view,
+                                         const ParticleTrackView& pt_view);
+
+    //@{
+    //! Accessors
+    CELER_FUNCTION bool on_boundary() const { return on_boundary_; }
+    CELER_FUNCTION real_type step() const { return step_; }
+    CELER_FUNCTION const OdeState& state() const { return state_; }
+    CELER_FUNCTION real_type safety() const { return safety_; }
+    CELER_FUNCTION Real3 origin() const { return origin_; }
```

Comment:
This can be discussed further even though this safety origin may not be used anywhere else  outside the field propagation in which the step may be sub-divided, if necessary, and the track is moved to an intermediate state where the safety is query to ensure whether it is inside the volume or not. If so, the safety origin is updated for the next trial stub-step within the current volume and so on.   In the linear navigator, the origin is the current position and never get updated within the current volume.